### ISSUE TITLE
Reintroducing API support should not cause a warning

### ISSUE THREAD

**Issue by terrajobst** [dotnet/roslyn-analyzers#4012]
### Repo

```C#
using System.Runtime.Versioning;

[assembly: SupportedOSPlatform("windows10.0")]

static class Program
{
    public static void Main()
    {
        SomeWindowsSpecific.Api();
    }
}

class SomeWindowsSpecific
{
    [SupportedOSPlatform("windows")]
    [UnsupportedOSPlatform("windows8.1")]
    [SupportedOSPlatform("windows10.0")]
    public static void Api() { }
}
```

### Expected

No diagnostics.

### Actual

> 'Api' is not supported or has been removed since windows 8.1 version

**Comment by jeffhandley** [comment]
This also repros with the guard methods being used instead of the assembly attribute:

```csharp

static class Program
{
    public static void Main()
    {
        if (OperatingSystem.IsWindowsVersionAtLeast(10))
        {
            WindowsSpecificApis.WorksIn10Plus();
        }
    }
}

class WindowsSpecificApis
{
    public static void WorksInAllVersions() { }

    [UnsupportedOSPlatform("windows")]
    [SupportedOSPlatform("windows10")]
    public static void WorksIn10Plus() { }
}
```

### Expected
No diagnostics.

### Actual
>  'WorksIn10Plus' is not supported or has been removed from windows

**Comment by buyaa-n** [comment]
Fixed and the [test added](https://github.com/dotnet/roslyn-analyzers/pull/3917/commits/1eba4b85b49c296650db237a417004ed37694f66#diff-aa2dc7c8b8aa7394d45b1352f85abfbbR762-R787), Thanks!
@jeffhandley's test also [added](https://github.com/dotnet/roslyn-analyzers/pull/3917/commits/1eba4b85b49c296650db237a417004ed37694f66#diff-143b956da2811185e4067197cdaed6fcR369-R397), but the call site need OS defined to be suppressed as API starts with Unsupported

**Comment by buyaa-n** [comment]
Closing by https://github.com/dotnet/roslyn-analyzers/pull/3917
